src/compiler.rs

```diff
@@ -106,7 +75,8 @@ pub fn compile_sync(
   parent_state: &Arc<IsolateState>,
   specifier: &str,
   referrer: &str,
-) -> Option<CodeFetchOutput> {
+  code_fetch_output: &mut ModuleMetaData,
+) -> () {
```

Comment:
Hmm... None seems like a clear way to indicate 404.

In this setup the code_fetch_output will just have empty fields... Why is this better?

Reply:
I am not sure you will ever try to compile a 404. Right?

Reply:
The "better" is that we a reusing the previous response, versus having the compiler send back a complete new object. I could return a new object in Rust that includes the previous data instead of mutating the old one.

Reply:
> I am not sure you will ever try to compile a 404. Right?

You shouldn't... but i think the function should be robust to it. The function as it is now, is a clear input -> output situation. However with this signature change it becomes much less clear what is input and output. I think this should change back.

Otherwise LGTM - im happy to land this before #1754 if that's easier for you.

Reply:
Ok, I will address it. 👍 I wouldn't wait for this with #1754, because it might take me longer.  I will try to do it quickly, but my other life has its demands which might not make it timely enough.